sdk/textanalytics/Azure.AI.TextAnalytics/src/HealthcareOperation.cs

```diff
@@ -206,17 +222,23 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     {
                         // we need to first assign a vaue and then mark the operation as completed to avoid race conditions
                         _value = Transforms.ConvertToRecognizeHealthcareEntitiesResultCollection(update.Value.Results, _idToIndexMap);
-
+                        Status = JobStatus.Succeeded;
                         NextLink = update.Value.NextLink;
                         _hasCompleted = true;
                     }
                     else if (update.Value.Status == JobStatus.Failed)
                     {
                         _requestFailedException = await ClientCommon.CreateExceptionForFailedOperationAsync(async, _diagnostics, _response, update.Value.Errors)
                             .ConfigureAwait(false);
+                        Status = JobStatus.Failed;
                         _hasCompleted = true;
                         throw _requestFailedException;
                     }
+                    else if (update.Value.Status == JobStatus.Cancelled)
+                    {
+                        Status = JobStatus.Cancelled;
+                        _hasCompleted = true;
+                    }
```

Comment:
Following the new .NET guideline => https://azure.github.io/azure-sdk/dotnet_implementation.html#dotnet-implement-operation we can follow what the "fail" path is doing. 
Create the `__requestFailedException` here and throw it. 
That way you don't need to change the logic on `Value`